src/Microsoft.IdentityModel.Tokens/BaseConfigurationManager.cs

```diff
@@ -35,9 +41,27 @@ public TimeSpan AutomaticRefreshInterval
                     throw LogHelper.LogExceptionMessage(new ArgumentOutOfRangeException(nameof(value), LogHelper.FormatInvariant(LogMessages.IDX10108, LogHelper.MarkAsNonPII(MinimumAutomaticRefreshInterval), LogHelper.MarkAsNonPII(value))));
 
                 _automaticRefreshInterval = value;
+                Interlocked.Exchange(ref _automaticRefreshIntervalInSeconds, value.TotalSeconds);
             }
         }
 
+        /// <summary>
+        /// Records the time this instance was created.
+        /// Used to determine if the automatic refresh or request refresh interval has passed.
+        /// The set is used for testing purposes.
+        /// </summary>
+        internal DateTimeOffset StartupTime { get; set; } = DateTimeOffset.UtcNow;
```

Comment:
I think it would help to be more explicit in spelling out the algorithm and how things end up relative to this.

Reply:
This is analogous to the seconds since the Unix epoch, so I think mentioning that would help (`BaseConfigurationManager` epoch).